### PR TITLE
Add all-in-one pass pipeline

### PR DESCRIPTION
**Pull request by Menooker**

### CODE REVIEW

**Comment by Menooker** [comment]
WIP: Depending on https://github.com/intel/graph-compiler/pull/70 and support of openmp in gcc.

File: lib/gc/Transforms/Pipeline.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+void populateCPUPipeline(mlir::PassManager &pm) {
+  // front-end, oneDNN graph dialect
+  populateFrontendPasses(pm);
+  // middle-end, LinalgX/Linalg/tensor dialects
+  populateTensorPasses(pm);
+  // middle-end, arith/math/vector dialects
+  populateVectorPasses(pm);
+  // back-end, arith/math/vector/memref dialects
+  populateBufferizationPasses(pm);
+  // REMOVE this pass after the TensorPasses are added. Currently we add this
```
Comment (ciyongch) [inline lib/gc/Transforms/Pipeline.cpp:139]: Can we add "TODO" to highlight the remaining items to better distinguish with other common comments?

File: src/gc-cpu-runner/gc-cpu-runner.cpp [hunk 2]
```diff
@@ -27,7 +27,11 @@
 #include "llvm/Support/TargetSelect.h"
 #include <stdio.h>
 
+extern int gc_runtime_keep_alive;
+
 int main(int argc, char **argv) {
+  // keeps GCCPURuntime linked
+  gc_runtime_keep_alive = 0;
```
Comment (ciyongch) [inline src/gc-cpu-runner/gc-cpu-runner.cpp:34]: Do we still need this?

File: test/gc/Transforms/Pipeline/run.mlir [hunk 3]
```diff
@@ -0,0 +1,23 @@
+// RUN: gc-opt %s --gc-cpu-pipeline | gc-cpu-runner -e main -entry-point-result=void | FileCheck %s
```
Comment (ciyongch) [inline test/gc/Transforms/Pipeline/run.mlir:1]: We might need to add some complex example to cover the default pipeline.

**Review by kurapov-peter — APPROVED** [review]
Looks good, could you please clean up all the commented out code before merge?

**Comment by kurapov-peter** [comment]
@AndreyPavlenko, please take a look

**Comment by AndreyPavlenko** [comment]
I think it would be nice to use the same code style for the entire code. Currently, one part of the code is in the CamelCase, another - in snake_case.

File: test/gc/Transforms/Pipeline/run.mlir (hunk 3 above)
Comment (Menooker) [reply test/gc/Transforms/Pipeline/run.mlir:1]: @LongshengDu Please help to add some test cases and check if the pipeline works for the current linalgx and onednngraph dialect. Maybe in another PR, or in this one?

**Comment by Menooker** [comment]
> Can we add "TODO" to highlight the remaining items to better distinguish with other common comments?

OK, changed the comments in pipeline to "todo"s.

> Looks good, could you please clean up all the commented out code before merge?

I have changed most of them to "todo"s. They mark the position where not-yet-implemented MLIR passes should be. The developers of that pass can know where to add to the pipeline in the future.

**Pull request merged**